File: python-version/LC3477/num_of_unplaced_fruits.py
from typing import List


class Solution:
    @staticmethod
    def numOfUnplacedFruits(fruits: List[int], baskets: List[int]) -> int:
        ans, baskets_length = len(fruits), len(baskets)
        flag = [False] * baskets_length
        for _, fruit in enumerate(fruits):
            for i, v in enumerate(baskets):
                if not flag[i] and fruit <= v:
                    flag[i] = True
                    ans -= 1
                    break
        return ans

File: go-version/leetcode/LC3477/num_of_unplaced_fruits.go
package LC3477

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/8/5 下午11:08
 * @FilePath: num_of_unplaced_fruits
 * @Description: 二层循环双重遍历
 */

func numOfUnplacedFruits(fruits, baskets []int) int {
	ans, basketLength := len(fruits), len(baskets)
	flag := make([]bool, basketLength)
	for _, fruit := range fruits {
		for i, v := range baskets {
			if !flag[i] && fruit <= v {
				flag[i] = true
				ans--
				break
			}
		}
	}
	return ans
}
